var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -0,0 +1,89 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Warpx(MakefilePackage):
+    """WarpX is an advanced electromagnetic Particle-In-Cell code. It supports
+    many features including Perfectly-Matched Layers (PML) and mesh refinement.
+    In addition, WarpX is a highly-parallel and highly-optimized code and
+    features hybrid OpenMP/MPI parallelization, advanced vectorization
+    techniques and load balancing capabilities.
+    """
+
+    homepage = "https://ecp-warpx.github.io/index.html"
+    url      = "https://bitbucket.org/berkeleylab/warpx.git"
+
+    version('master', git='https://bitbucket.org/berkeleylab/warpx.git', tag='master')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('gcc')
+
+    variant('dims',
+        default='3',
+        values=('1', '2', '3'),
+        multi=False,
+        description='Number of spatial dimensions')
+
+    variant('psatd', default=False, description='Enable PSATD solver')
+    variant('do_electrostatic', default=False, description='Include electrostatic solver')
+    variant('debug', default=False, description='Enable debugging features')
+    variant('tprof', default=False, description='Enable tiny profiling features')
+    variant('openmp', default=True, description='Enable OpenMP features')
+
+    resource(
+       name='amrex',
+       git='https://github.com/AMReX-Codes/amrex.git',
+       tag='development',
+       destination='.'
+    )
+
+    resource(
+       name='picsar',
+       git='https://dpgrote@bitbucket.org/berkeleylab/picsar.git',
```

Comment:
this git URI should use a public URL with anonymous checkout if possible. Otherwise other people can not check it out.